### DISCUSSION TITLE
Agent for granite 3/ llama 3 model running on OpenShift

### DISCUSSION THREAD

**Discussion by gskumar1010** [i-am-bee/beeai-framework#192] [Q&A]
If there is a granite3/ llama 3 model running on OpenShift/Kubernetes,

Is there a way of building agent  using the Bee framework?

Thanks

**Comment by gskumar1010** [comment]
Just to clarify:

If I have the agent running on OpenShift, how can it interact with a model that is also running on OpenShift?

https://github.com/i-am-bee/bee-agent-framework/blob/main/.env.template#L9
expects Watsonx details. Is there a way of just providing the url of the LLM thats being served on OpenShift?

Thanks

**Reply by Tomas2D** [reply]
In that case, I assume you would like to use a vLLM adapter.

**Comment by gskumar1010** [comment]
Thanks @Tomas2D for the response. A  couple of quick  questions.
1) Can we use the a vllm adapter like [ibm-vllm](https://github.com/i-am-bee/bee-agent-framework/blob/b5e10fa37051e5cb1c55bc7cba0e20c00a405a05/examples/llms/providers/ibm-vllm.ts) or do you see  a need to implement a custom vllm adapter
2) So, if the llm is running on OpenShift on vllm, is there a way of providing the url?
[vllm-adapter](https://github.com/i-am-bee/bee-agent-framework/blob/b5e10fa37051e5cb1c55bc7cba0e20c00a405a05/examples/llms/providers/ibm-vllm.ts#L12) seems to just have option for modelId

**Reply by Tomas2D** [reply]
See https://github.com/i-am-bee/bee-agent-framework/blob/main/examples/llms/providers/ibm-vllm.ts.

The `Client` class takes URL which is then processed. See implementation https://github.com/i-am-bee/bee-agent-framework/blob/b5e10fa37051e5cb1c55bc7cba0e20c00a405a05/src/adapters/ibm-vllm/client.ts#L91-L110
